packages/type-utils/src/TypeOrValueSpecifier.ts

```diff
@@ -0,0 +1,202 @@
+import path from 'path';
+import type * as ts from 'typescript';
+
+interface FileSpecifier {
+  from: 'file';
+  name: string | Array<string>;
+  source?: string;
```

Comment:
You know, now I'm having second thoughts 😅 🤔 ... maybe it should be `source: 'file'`, and `from?: string`? WDYT?

(at this point I've gone back and forth so many times, I'm ok with either way...)

Reply:
I think I would generally associate either `from` or `source` with the *type* of the item (package/file/lib). However, the `source` property as it is now is confusing IMHO mainly because it tries to use the same name for 2 concepts that generally don't overlap that much. So I would keep `from` and rename `source` to something like `path` for file specifier and `package` for package specifier. That I think is much clearer - yes, you need to use 2 different keys but I think "clarity > terseness"

Reply:
Hmmmm. You make a good point. Especially since we've eased back a bit on the terse multiple-formats-at-the-same-time style.

Paging @typescript-eslint/triage-team for input - I've looked at these options so many times I'm having a hard time being objective.

Reply:
`source` is a bit ambiguous IMO.
I think `path` and `packageName` are better.

The one question I have (because it's not clear to me) is `source` supposed to be relative or absolute? If it's relative - relative to what?

We need to be clear on this in both our code and the docs.

Reply:
Hmm I like `path` and `package`... just updated it now. Time to merge! Thank you!